toolchain/base/value_ids.h

```diff
@@ -71,6 +71,43 @@ struct IdentifierId : public IdBase<IdentifierId> {
 };
 constexpr IdentifierId IdentifierId::None(IdentifierId::NoneIndex);
 
+// The name of a package, which is either an identifier or the special `Core`
+// package name.
+//
+// TODO: Consider also treating `Main` and `Cpp` as special package names.
+struct PackageNameId : public IdBase<PackageNameId> {
```

Comment:
Why does this end up in `base` instead of `sem_ir`?

Reply:
It's also used by `Parse`, because that's where we first parse and check `package` and `import` declarations. I could move it to `Parse` if you prefer, but it's *mostly* a check-level thing so putting it in `Parse` felt a bit odd. Alternatively, I could put it in `Parse` and add a `using` to also pull it into `Check`. What would you prefer?

Reply:
I was overlooking the `tree.h` use. It makes me wonder though, I'd made the library name a `StringLiteralValueId` instead of a `LibraryNameId` in order to abstract that part out. If you're wanting to head more in this direction in order to make passing `PackageNameId::Core` straightforward, should there be a TODO somewhere (tree.h on `library_id` or ids.h on `LibraryNameId`) to move `LibraryNameId` here and use it in parse for consistency?

Reply:
TODO added.